### ISSUE TITLE
Revise bundling strategy

### ISSUE THREAD

**Issue by TyDunn** [continuedev/continue#320]
We have wanted to do this, and I keep hear it coming up from users:
- "You may want to revise your bundling strategy because this is quite slow on a beefy machine"
- https://github.com/continuedev/continue/issues/300#issuecomment-1654994274
- "Also, it takes a long time to load every time, downloading python packages. Are you caching them so that it doesn't download every library from scratch every time?"

**Comment by sestinj** [comment]
@bhaktatejas922 and others that care about slow initial install: I've taken an first look at ways of speeding it up:
- use the pip `--no-compile` flag, which will cause compilation of python code to bytecode to happen at runtime, essentially spreading out the work to runtime. This might give faster perceived loading time.
- Use one of the following packaging tools
    - cx_freeze
    - pyinstaller
    - PyOxidizer
    - pex
    
I'll be trying to implement one of these in the next couple days. If anyone comes across this issue and has experience with these tools, I'll be open to feedback on which works best.

**Comment by bhaktatejas922** [comment]
I've used pyinstaller and nuitka before. Pyinstaller is simpler but nuitka runs faster that default python cause it compiles some thing to lower level languages. Nuitka depends on libc and some other stuff though

**Comment by sestinj** [comment]
Good to know. Any catches to pyinstaller that might not be immediately apparent?

**Comment by bhaktatejas922** [comment]
nope, should be straightforward.

**Comment by sestinj** [comment]
An update on this issue's progress:
- Pyinstaller works just great, and I've been able to set up a GitHub workflow to package for linux, windows, and macos, and then run the correct executable at runtime from the VS Code extension.
- But it causes issues with importing `config.py`, because we are now running a frozen module from Pyinstaller. Using importlib, we can import config.py, but then imports _inside_ of config.py are broken.
- (happening in sdk.py) So far, I've attempted both `exec` and `importlib.util.module_from_spec` -> `spec.loader.exec_module` in order to load config.py, both of which fail when trying to import modules that I've tested can be successfully imported from sdk.py itself (e.g. config.py tries to import anthropic, and this fails, but I can import anthropic inside of sdk.py)
- One (slightly messy) attempt at solving this involves catching ModuleNotFound errors, and then manually importing them inside sdk.py before re-executing config.py. This works if I import manually in sdk.py, for example `from anthropic import AsyncAnthropic`, but _not_ if I use `importlib.import_module`. Pausing for a minute, but so far unsure why these two don't act the same. The latter would be necessary to dynamically catch the imports from the error.
